src/render/metal/SDL_render_metal.m

```diff
@@ -2024,6 +2043,7 @@ - (void)dealloc
         if (data.mtllayer.displaySyncEnabled) {
             renderer->info.flags |= SDL_RENDERER_PRESENTVSYNC;
         }
+        renderer->SetVSync = METAL_SetVSync;
```

Comment:
Should probably assign this unconditionally, so that we can have the error message be more specific when unsupported.